tests/always/test_project_structure.py

```diff
@@ -213,76 +299,110 @@ class TestGoogleProviderProjectStructure(unittest.TestCase):
+        'airflow.providers.google.cloud.operators.vision.CloudVisionDeleteProductSetOperator',
+        'airflow.providers.google.cloud.operators.vision.CloudVisionDeleteReferenceImageOperator',
+        'airflow.providers.google.cloud.operators.workflows.WorkflowsDeleteWorkflowOperator',
+        'airflow.providers.google.marketing_platform.sensors.campaign_manager.'
+        'GoogleCampaignManagerReportSensor',
+        'airflow.providers.google.marketing_platform.sensors.display_video.'
+        'GoogleDisplayVideo360GetSDFDownloadOperationSensor',
+        'airflow.providers.google.marketing_platform.sensors.display_video.'
+        'GoogleDisplayVideo360ReportSensor',
+        'airflow.providers.google.marketing_platform.sensors.search_ads.GoogleSearchAdsReportSensor',
+    }
 
-        all_operators -= self.MISSING_EXAMPLES_FOR_OPERATORS
-        all_operators -= self.DEPRECATED_OPERATORS
-        all_operators -= self.BASE_OPERATORS
-        assert set() == all_operators
-
-    @parameterized.expand(
-        itertools.product(["_system.py", "_system_helper.py"], ["operators", "sensors", "transfers"])
-    )
-    def test_detect_invalid_system_tests(self, resource_type, filename_suffix):
-        operators_tests = self.find_resource_files(top_level_directory="tests", resource_type=resource_type)
-        operators_files = self.find_resource_files(top_level_directory="airflow", resource_type=resource_type)
-
-        files = {f for f in operators_tests if f.endswith(filename_suffix)}
-
-        expected_files = (f"tests/{f[8:]}" for f in operators_files)
-        expected_files = (f.replace(".py", filename_suffix).replace("/test_", "/") for f in expected_files)
-        expected_files = {f'{f.rpartition("/")[0]}/test_{f.rpartition("/")[2]}' for f in expected_files}
-
-        assert set() == files - expected_files
-
-    @staticmethod
-    def find_resource_files(
-        top_level_directory: str = "airflow",
-        department: str = "*",
-        resource_type: str = "*",
-        service: str = "*",
-    ):
-        python_files = glob.glob(
-            f"{ROOT_FOLDER}/{top_level_directory}/providers/google/{department}/{resource_type}/{service}.py"
-        )
-        # Make path relative
-        resource_files = (os.path.relpath(f, ROOT_FOLDER) for f in python_files)
-        # Exclude __init__.py and pycache
-        resource_files = (f for f in resource_files if not f.endswith("__init__.py"))
-        return resource_files
 
-    @staticmethod
-    def examples_for_service(service_name):
-        yield from glob.glob(
-            f"{ROOT_FOLDER}/airflow/providers/google/*/example_dags/example_{service_name}*.py"
-        )
-        yield from glob.glob(f"{ROOT_FOLDER}/tests/system/providers/google/{service_name}/example_*.py")
+class TestElasticsearchProviderProjectStructure(ExampleCoverageTest):
+    PROVIDER = "elasticsearch"
+    OPERATOR_DIRS = {"hooks"}
```

Comment:
The verbiage of the new code above is very Operator-centric, but this provider already breaks that naming paradigm (as well as Sensors, and Transfers). I wonder if something more generic makes sense? Maybe just `DIRS`? And similar changes to genericize the new classes above.

Reply:
I agree, I'm not satisfied with current naming (just took over the old naming). Same goes for method names such as list_of_operators etc. I will think about it

Reply:
Used "class" generic name. CLASS_DIRS, list_of_classes, class_paths, DEPRECATED_CLASSES, MISSING_EXAMPLES_FOR_CLASSES etc